### PR TITLE
the_efficiency_of_the_hydraulic_press_from_force_and_height

### PR DESCRIPTION
**Pull request by Darya2909**

### CODE REVIEW

File: symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+                           validate_output, dimensionless)
+
+# Description
+## Since the hydraulic press is a mechanism, its operation can be characterized by a coefficient of efficiency.
+
+## Law: n = (F2 * h2) / (F1 * h1)
+## Where:
+## F2 is the force spent on useful work
+## h2 is the useful movement
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py:11]: Useful movement sounds weird.

File: symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from sympy import (Eq, solve)
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+                           validate_output, dimensionless)
+
+# Description
+## Since the hydraulic press is a mechanism, its operation can be characterized by a coefficient of efficiency.
+
+## Law: n = (F2 * h2) / (F1 * h1)
```
[the remark this answers is not in the document]
Comment (Darya2909) [reply symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py:8]: https://educon.by/index.php/materials/phys/gidrostatika

File: symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+from sympy import (Eq, solve)
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py:1]: Remove `the` from file name for shorter text.

File: symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+## Law: n = (F2 * h2) / (F1 * h1)
+## Where:
+## F2 is the force spent on useful work (work on lifting the load)
+## h2 is the movement of load
+## F1 is expended force
+## h1 is expended height
+## n is coefficient of efficiency
+
+## Conditions
+## The efficiency is calculated if we are talking about an imperfect hydraulic press.
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py:17]: Remove. For ideal press efficiency is 1 - it's OK.

File: symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+law = Eq(efficiency, (useful_force * useful_height) / (expended_force * expended_height))
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(useful_force_=useful_force, useful_height_=useful_height, expended_force_=expended_force, expended_height_=expended_height)
+@validate_output(efficiency)
+def calculate_efficiency(useful_force_: Quantity, useful_height_, expended_force_, expended_height_: Quantity) -> Quantity:
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/the_efficiency_of_the_hydraulic_press_from_force_and_height.py:36]: Return `float` instead `Quantity`.
See other laws for example of converting to float.

File: test/hydro/efficiency_of_the_hydraulic_press_from_force_and_height_test.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    return Args(
+        useful_force=useful_force,
+        useful_height=useful_height,
+        expended_force=expended_force,
+        expended_height=expended_height
+               )
+
+
+def test_basic_efficiency(test_args):
+    result_volume = efficiency.calculate_efficiency(test_args.useful_force, test_args.useful_height, test_args.expended_force, test_args.expended_height)
```
Comment (blackyblack) [inline test/hydro/efficiency_of_the_hydraulic_press_from_force_and_height_test.py:31]: It's not volume.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**